### PR TITLE
Enable OpenSSF Scorecard Github Action and Badge

### LINKED ISSUE #37343: Enable OpenSSF Scorecard Github Action and Badge
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed feature requests
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Proposal

Hi, I'm Joyce from Google and I'm working on behalf of the [Open Source Security Foundation][ossf] (OpenSSF) to help essential open-source projects improve their supply-chain security.

The OpenSSF has developed, [in partnership with GitHub][sc-blog], a tool called [Scorecards][sc]. Scorecards runs dozens of automated security [checks][checks] to help maintainers better understand their project's supply-chain security posture. 

I would like to suggest the adoption of the [Scorecard GitHub Action][sc-gha], which was developed by the OpenSSF to make it easier to run the Scorecard checks on projects hosted in Github. The action is very lightweight and runs on every change to the repository's main branch. 

[checks]: https://github.com/ossf/scorecard#scorecard-checks
[ossf]: https://openssf.org/
[sc]: https://github.com/ossf/scorecard
[sc-blog]: https://github.blog/2022-01-19-reducing-security-risk-oss-actions-opensff-scorecards-v4/
[sc-gha]: https://github.com/ossf/scorecard-action

### Motivation and context

The results of the Scorecard's checks are available on the project's [security dashboard](https://github.com/twbs/bootstrap/security), and include suggestions on how to solve any issues (see examples below). The Action does not run or interact with any workflows, but merely parses them to identify possible vulnerabilities. This Action has been adopted by 1800+ projects already, having some prominent users like [Tensorflow][tensorflow], [Angular][angular], [Flutter][flutter], [sos.dev][sos-dev] and [deps.dev][deps-dev].

The Bootstrap project is already following most of the supply-chain security best practices (the top 6.6% greatest scorecard scores), but the Scorecard would still help to track other security posture improvements, to guarantee the already followed ones would still be followed, to be up to date to new security best practices since Scorecard is in continuous improvement and, if you opted for the [badge][badge], it would also show to the users the project's commitment to security best practices.   

Would you be interested in a PR which adds this Action? Optionally, it can also publish your results to the OpenSSF REST API, which allows the previously mentioned [badge][badge] with the project's score to be added to the README file.

In case of doubts or concerns you can check out the [Scorecards FAQ][faq]. Anyway, feel free to reach me out.

![Code scanning dashboard with multiple alerts, including Code-Review and Token-Permissions][img-security]

![Detail of a Token-Permissions alert, indicating the specific file and remediation steps][img-detail]

[checks]: https://github.com/ossf/scorecard#scorecard-checks
[badge]: https://openssf.org/blog/2022/09/08/show-off-your-security-score-announcing-scorecards-badges/
[faq]: https://github.com/ossf/scorecard/blob/main/docs/faq.md#frequently-asked-questions 
[img-security]: https://user-images.githubusercontent.com/15221358/190184391-84ca1844-259a-4b3b-9c86-74adadbea7f1.png
[img-detail]: https://user-images.githubusercontent.com/15221358/190184600-ee8d3b39-077e-416a-8711-1b5fb01cf0b3.png

[tensorflow]: https://github.com/tensorflow/tensorflow
[angular]: https://github.com/angular/angular  
[flutter]: https://github.com/flutter/flutter 
[sos-dev]: https://sos.dev/ 
[deps-dev]: https://deps.dev/

### PR DESCRIPTION
**Pull request by joycebrum**
Signed-off-by: Joyce Brum <joycebrum@google.com>
CC @XhmikosR 

### Description

- Scorecards.yml file enables the Scorecard action to run on push to main and once a week (important for some checks like contribution check)
- Readme file with badge from shields.io

Any doubts or concerns please let me know.

### Motivation & Context

Closes #37343

It enables the OpenSSF Scorecard Github Action and Badge to help you to ensure the project will continue to follow the open source best practices or even improve any possible practice to avoid security risks and vulnerabilities. 

### Type of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would change existing functionality)

### Checklist

- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)
- [X] My code follows the code style of the project _(using `npm run lint`)_
- [X] My change introduces changes to the documentation
- [X] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] All new and existing tests passed

#### Live previews

https://github.com/joycebrum/bootstrap#status

### Related issues

#37343

### CODE REVIEW

File: .github/workflows/scorecards.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      security-events: write
+      # Used to receive a badge.
+      id-token: write
+      # Needs for private repositories.
+      contents: read
+      actions: read
+
+    steps:
+      - name: "Checkout code"
+        uses: actions/checkout@a12a3943b4bdde767164f792f33f40b04645d846 # tag=v3.0.0
```
Comment (XhmikosR) [inline .github/workflows/scorecards.yml:28]: Can we use the tags when possible?

File: README.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ Read the [Getting started page](https://getbootstrap.com/docs/5.2/getting-starte
 [![BrowserStack Status](https://www.browserstack.com/automate/badge.svg?badge_key=SkxZcStBeExEdVJqQ2hWYnlWckpkNmNEY213SFp6WHFETWk2bGFuY3pCbz0tLXhqbHJsVlZhQnRBdEpod3NLSDMzaHc9PQ==--3d0b75245708616eb93113221beece33e680b229)](https://www.browserstack.com/automate/public-build/SkxZcStBeExEdVJqQ2hWYnlWckpkNmNEY213SFp6WHFETWk2bGFuY3pCbz0tLXhqbHJsVlZhQnRBdEpod3NLSDMzaHc9PQ==--3d0b75245708616eb93113221beece33e680b229)
 [![Backers on Open Collective](https://img.shields.io/opencollective/backers/bootstrap)](#backers)
 [![Sponsors on Open Collective](https://img.shields.io/opencollective/sponsors/bootstrap)](#sponsors)
-
+[![OpenSSF Scorecard](https://img.shields.io/ossf-scorecard/github.com/twbs/bootstrap)](https://api.securityscorecards.dev/projects/github.com/twbs/bootstrap)
```
Comment (XhmikosR) [inline README.md:75]: Isn't there any other human readable link we could use here?

File: .github/workflows/scorecards.yml (hunk 1 above)
Comment (XhmikosR) [reply .github/workflows/scorecards.yml:28]: BTW this is one of the things I disagree with the scorecard rules. I know the pros and cons, and I'd rather have a tag for actions than have another thing to keep up to date even if it's with dependabot. Too much noise.

File: README.md (hunk 2 above)
Comment (joycebrum) [reply README.md:75]: The Scorecard team is working on improving this output https://github.com/ossf/scorecard-webapp/issues/206, right now it is really far from good. 

If you rather wait for it to add the link I can remove it

**Comment by XhmikosR** [comment]
@mdo: What's your take on this?

I removed the link since there isn't a human readable page to link to.

Overall, the checks are OK, I mostly disagree with the Actions pinning for the reasons I mentioned above. The other things are generally considered good practices.

That being said, 7.2 is not good enough either and it's mostly caused by Actions not being pinned AFAICT.

**Comment by XhmikosR** [comment]
BTW, maybe we could link to https://bestpractices.coreinfrastructure.org/en/projects/2930 and add the best practices badge? It seems @bardiharborow set this up, not sure how to get updated results or access ourselves.

**Comment by joycebrum** [comment]
About wheter the Pinned-Dependencies has a big weight in the current score, I've made some calculations:

“Critical” risk checks are weighted at 10
“High” risk checks are weighted at 7.5
“Medium” risk checks are weighted at 5
“Low” risk checks are weighted at 2.5
This we would have:

( 10 * (Dangerous-Workflow) + 7.5 * (Binary-Artifacts + Branch-Protection + Code-Review + Dependency-Update-Tool + Maintained + Signed-Releases + Token-Permissions + Vulnerabilities) + 5 * (Fuzzing + Pinned-Dependencies + <s>Packaging</s> + SAST + Security-Policy) + 2.5 * (CI-Tests + CII-Best-Practices + Contributors + License) ) / (10 + 60 + 20 + 10)

Then, considering the current score:

$$ (100 + 7.5*(10 + 8 + 9 + 10 + 10 + 0 + 0 + 10) + 60 + 75) + 5 * (0 + 7 + 10 + 10) + 2.5 * (10 + 2 + 10 + 10)) / 100 $$

$$ (100 + 7.5 *  57 + 5 * 27 + 2.5 * 32 ) /100 = 7.4 $$

Even if the Pinned Dependencies was a 10, the score would only go to 7.6.

According this, the best approach would be:

1. Signed-Releases
2. Token-Permissions (Quick-win)
3. Fuzzing
4. CII-Best-Practices

Working only to get this four to 10 would increase the score to 9,6

**Comment by bardiharborow** [comment]
@XhmikosR it says that everyone with commit access to the repository should have access already, but if not, I can manually add user IDs to the access list.

**Comment by XhmikosR** [comment]
I'm going to give this a go and we revert it if we don't like it later.

**Review by XhmikosR — APPROVED** [review]

**Pull request merged**